### PR TITLE
fix: quick buy infinite loading

### LINKED ISSUE #9477: quick buy not loading
its fast on production, but i struggle on canary 

https://github.com/kodadot/nft-gallery/assets/90852205/44136d06-8bcb-4529-9390-0f5ecaa702dc

image
idk if this error helps

### PR DESCRIPTION
**Pull request by pbkompasz**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [X] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [X] Closes #9477
- [X] Related #7051
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [X] My contribution builds **clean without any errors or warnings**
- [X] I've merged recent default branch -- **main** and I've no conflicts
- [X] I've tried to respect high code quality standards
- [X] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [X] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt&usdamount=50&donation=true)

#### Community participation

- [ ] [Are you at KodaDot Ecosystem Telegram?](https://t.me/kodadot_eco)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### CODE REVIEW

**Comment by pbkompasz** [comment]
Test is failing due to `Commands.e2elogin()`

File: components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue [hunk 1]
```diff
@@ -5,6 +5,7 @@
     scroll="clip"
     class="z-[1000]"
     content-class="modal-width"
+    data-testid="confirm-modal"
```
Comment (prury) [inline components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue:8]: for some reason i could not find this data-testid on the generated page

File: tests/e2e/buy.spec.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { expect, test } from './fixtures'
+
+const ITEMS_PATH = '/ahp/explore/items?listed=true'
+
+test('Check buy now option', async ({ page, Commands }) => {
+  await Commands.e2elogin()
+  await page.goto(ITEMS_PATH)
+
+  await test.step('Buy now from explorer', async () => {
+    await page.getByTestId('item-buy').first().click()
```
Comment (prury) [inline tests/e2e/buy.spec.ts:10]: you have to hover over the first card for the buy now button to appear

File: tests/e2e/buy.spec.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    await expect(await page.getByTestId('confirm-modal')).toBeVisible()
```
Comment (prury) [inline tests/e2e/buy.spec.ts:11]: we may have to add timeout, since its not always fast

**Comment by prury** [comment]
> Test is failing due to `Commands.e2elogin()`

nope, its actually the test itself, lacking some lines, appreciate it tho

regarding solution, buy now no longer on infinite loading ✔️

**Comment by hassnian** [comment]
@pbkompasz this section

CleanShot 2024-02-20 at 09 01 15@2x

 it's working at it should, so no changes needed there

the issue is that , when trying to subscribe for price changes , it's failing to connect to the websocket 

@kodadot/internal-dev  can you guys check that ? 
 
CleanShot 2024-02-20 at 08 59 34@2x

handle the onError case inside `subscribeToNftPriceChange`'s  graphql subscription.

thanks

**Comment by vikiival** [comment]
Cc @preschian

**Comment by preschian** [comment]
> the issue is that , it's failing to connect to the websocket when trying to subscribe for price changes ,
> 
> @kodadot/internal-dev can you guys check that ? `wss://ahp.gql.api.kodadot.xyz/`

oooh, this one, when trying to target subscriptions, should use old endpoints. @hassnian @pbkompasz could you check if this file is working or not https://github.com/kodadot/nft-gallery/blob/main/utils/websocket.ts#L4-L8

recently I changed that files

File: components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue [hunk 4]
```diff
@@ -104,7 +105,14 @@ const isDisabledPurchasePrefix = isRemark
...
 
-const loading = computed(() => !autoteleport.value?.isReady || props.loading)
+const loading = ref(props.loading)
+
+watch(
+  () => autoteleport.value?.isReady,
+  (val) => {
+    loading.value = !val
+  },
+)
```
Comment (daiagi) [inline components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue:115]: How about
```suggestion
const loading = computed(() => autoteleport.value?.isReady ? !autoteleport.value.isReady : props.loading)
```

Comment (hassnian) [inline components/common/confirmPurchaseModal/ConfirmPurchaseModal.vue:115]: @daiagi no changes needed here 

this is the initial code 

CleanShot 2024-02-21 at 14 57 27@2x

I think it's does the same as your version 

the issue https://github.com/kodadot/nft-gallery/issues/9477 got already fixed by  #9491

https://github.com/kodadot/nft-gallery/pull/9484#issuecomment-1954109180


if anything, we can handle the error case inside the subscription 

CleanShot 2024-02-21 at 14 55 35@2x

**Comment by prury** [comment]
@hassnian close this one then?

**Comment by hassnian** [comment]
> @hassnian close this one then? 

Sure, since it's already working 

If anything as I said the only thing would be to handle the case where the websocket connection fails

**Pull request closed without merging**